content/en/docs/components/pipelines/sdk/v2/component-development.md

```diff
@@ -75,7 +75,7 @@ When you design your component's code, consider the following:
 
     _Parameters_ typically represent settings that affect the behavior of your pipeline.
     Parameters are passed into your component by value, and can be of any of
-    the following types: `int`, `double`, `float`, or `str`. Since parameters are
```

Comment:
I'm removing `double` as it's not a Python build-in type. We do recognize `"double"` (as a string) though, but it's more like variation of `float`, similarly as we treat `"integer"` as `int`, `"string"` and `"text"` as `str`.
I feel like keep this list shorter and cleaner (with all Python build-in types) is probably better than listing all variations. WDYT? @joeliedtke

Reply:
Also we treat `typing.Dict[...]` and `"JsonObject"` the same as `dict`, and treat `typing.List[...]` and `"JsonArray"` the same as `list`. Let me know if you think we should document all these variations.

Here's the almost complete list in our code: https://github.com/kubeflow/pipelines/blob/14cb98b662bc88a28122dcc5d4bf7d577c7b218f/sdk/python/kfp/dsl/type_utils.py#L35-L49
It's almost complete because it doesn't reflect that we treat `typing.Dict`, `typing.Dict[str, Any]`, etc. as `dict`.